### PR TITLE
Better use of AuthComponent sessionKey

### PR DESCRIPTION
**Pull request by efanzini**

### CODE REVIEW

File: src/Controller/Component/AuthComponent.php [hunk 1]
```diff
@@ -597,7 +597,7 @@ public function deny($actions = null)
     public function setUser(array $user)
     {
         $this->session->renew();
-        $this->session->write($this->sessionKey, $user);
+        $this->session->write($this->sessionKey . '.user', $user);
```
Comment (dereuromark) [inline src/Controller/Component/AuthComponent.php]: You are changing `U` to `u`. On purpose? in RC BC breaks should be avoided so close to the final release if possible.

Comment (efanzini) [inline src/Controller/Component/AuthComponent.php]: i changed it because all other keys are lowercase.
you should use Auth->user() to access user data

**Comment by ADmad** [comment]
This will break backwards compatibility and gives no real benefit so :-1:

**Comment by efanzini** [comment]
if we keep the capital 'U' there will be no compatibility break

**Comment by ADmad** [comment]
It will still cause issues for people using Auth->sessionKey in their app code.

File: src/Controller/Component/AuthComponent.php (hunk 1 above)
Comment (markstory) [inline src/Controller/Component/AuthComponent.php]: This needs to stay User. Repurposing the property will very likely break userland code too.

Comment (efanzini) [inline src/Controller/Component/AuthComponent.php]: It shouldn't. user data should be accessed via the provided method

http://book.cakephp.org/2.0/en/core-libraries/components/authentication.html#accessing-the-logged-in-user

**Comment by efanzini** [comment]
AuthComponent session just stores redirect url and logged user data.

redirect urls should be set using the config() method and user data should be accessed using user() method

i don't see any reason to access sessionKey directly

**Comment by ADmad** [comment]
> AuthComponent session just stores redirect url and logged user data.

I am well aware of what the component does :)

> i don't see any reason to access sessionKey directly

You don't but someone else might. It's a public property so people can use it in whatever way they see fit.

**Pull request closed without merging**